cosmoz-data-nav.js

```diff
@@ -730,17 +723,29 @@
 				return;
 			}
 
+			const selected = this.selected;
+			if (selected) {
+				const selectedIndex = queue.indexOf(selected);
+				if (selectedIndex > 0) {
+					queue.unshift(...queue.splice(selectedIndex, 1));
+				}
+			}
+
 			let renderRun = false,
 				reRun = true;
 
 			this._indexRenderQueue = queue
-				.sort((a, b) => a === this.selected ? -1 : b === this.selected ? 1 : 0)
 				.map(idx => {
 
 					const elementIndex = idx % this.elementsBuffer,
 						element = this._elements[elementIndex],
 						item = this.items[idx];
 
+					if (renderRun) {
```

Comment:
We should not do anything else after a render has run. Incomplete items might be complete in next run.

Reply:
I disagree, I think we should do whatever inexpensive task that is left in the queue but avoid the expensive ones.
Since `resetElement` won't reset a previously reset element, we should make sure that the elements in the buffer are in sync in case the user navigates back or forth to show the loading, since we won't render during animation.

Reply:
@nomego We set `renderRun` to `true` manually so that it returns the idx for the next items. If we want something more to run we should not set renderRun to true?

Reply:
Well it only prevents any _render_ task to run, it could still empty the queue of tasks setting incompelte status/toggling instances.